pkg/trie/trie.go

```diff
@@ -54,13 +63,21 @@ type Hashable interface {
 	Hash() (common.Hash, error)
 }
 
-type Trie interface {
-	PrefixTrie
-	KVStore
+type ReadOnlyTrie interface {
```

Comment:
I'm a little confused, why do we need separate read and write tries?

Reply:
We don't really need it but I created it to check that I'm just implementing the reads. We can rollback this change and go back to the previous interfaces

Reply:
I like this interface separation, gives us a better way to compose the implementation also to understand the implementation's responsibilities